flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperator.java

```diff
@@ -419,6 +435,14 @@ public void merge(W mergeResult,
 				registerCleanupTimer(window);
 			}
 		}
+
+		// side output input event if
+		// element not handled by any window
+		// late arriving tag has been set
+		// windowAssigner is event time and current timestamp + allowed lateness no less than element timestamp
+		if(isSkippedElement && lateDataOutputTag != null && isLate(element)) {
```

Comment:
I thought about this again. I think it doesn't hurt to have it because it catches the case when a `WindowAssigner` doesn't assign any windows. In that case an element is also "skipped" but it is not necessarily considered late. What do you think?